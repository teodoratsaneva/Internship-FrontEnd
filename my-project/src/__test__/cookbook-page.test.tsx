import { render } from "@testing-library/react";
import CookbookPage from "../pages/cookbook-page";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import HeaderComponent from "../common-components/header-page-component";
import RecipeComponent from "../recipe-card/recipe-form";
import { shallow } from "enzyme";

describe ('CookbookPage', () => {
    it('renders correctly Cookbook page', () => {
        const tree = render(
			<BrowserRouter>
				<CookbookPage />
			</BrowserRouter>
		);

        expect(tree).toMatchSnapshot();
    });

    it('should handle the case where there are no recipes in local storage', () => {
        localStorage.removeItem('items');
        const { queryAllByTestId } = render(
            <BrowserRouter>
                <CookbookPage />
            </BrowserRouter>
        );
    
        const recipesContainers = queryAllByTestId('recipes-container');
        expect(recipesContainers).toHaveLength(0);
    });

    it('should render a list of recipe components', () => {
        const setStateMock = jest.fn();
		const useStateMock: any = (useState: any) => [useState, setStateMock];
		jest.spyOn(React, "useState").mockImplementation(useStateMock);

        const recipes = [
          { id: '1', title: 'Recipe 1', ingredients: [] },
          { id: '2', title: 'Recipe 2', ingredients: [] },
          { id: '3', title: 'Recipe 3', ingredients: [] },
        ];
    
        const localStorageMock = {
          getItem: jest.fn(),
          setItem: jest.fn(),
          removeItem: jest.fn(),
        };
        global.localStorage = localStorageMock as any;
    
        localStorageMock.setItem.mockReturnValue(JSON.stringify(recipes));
    
        const wrapper = shallow(
          <BrowserRouter>
            <CookbookPage />
          </BrowserRouter>
        );
    
        expect(wrapper.find(HeaderComponent)).toHaveLength(1);
        expect(wrapper.find(RecipeComponent)).toHaveLength(2);
      });
})