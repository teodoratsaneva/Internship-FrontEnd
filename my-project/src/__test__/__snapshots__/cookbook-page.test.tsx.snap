// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookbookPage renders correctly Cookbook page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          href="/"
        >
          <img
            alt="darkLogo"
            class="darkLogo"
            src="darkLogo.JPG"
          />
        </a>
      </header>
      <div
        class="cookbook-page-container"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
        >
          Welcome to the Cookbook. Here is a list of your potion recipes
        </h1>
        <div
          class="recipes-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        href="/"
      >
        <img
          alt="darkLogo"
          class="darkLogo"
          src="darkLogo.JPG"
        />
      </a>
    </header>
    <div
      class="cookbook-page-container"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      >
        Welcome to the Cookbook. Here is a list of your potion recipes
      </h1>
      <div
        class="recipes-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders correctly cookbook page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          href="/"
        >
          <img
            alt="darkLogo"
            class="darkLogo"
            src="darkLogo.JPG"
          />
        </a>
      </header>
      <div
        class="cookbook-page-container"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
        >
          Welcome to the Cookbook. Here is a list of your potion recipes
        </h1>
        <div
          class="recipes-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        href="/"
      >
        <img
          alt="darkLogo"
          class="darkLogo"
          src="darkLogo.JPG"
        />
      </a>
    </header>
    <div
      class="cookbook-page-container"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      >
        Welcome to the Cookbook. Here is a list of your potion recipes
      </h1>
      <div
        class="recipes-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders correctly cookbook page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          href="/"
        >
          <img
            alt="darkLogo"
            class="darkLogo"
            src="darkLogo.JPG"
          />
        </a>
      </header>
      <div
        class="cookbook-page-container"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
        >
          Welcome to the Cookbook. Here is a list of your potion recipes
        </h1>
        <div
          class="recipes-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        href="/"
      >
        <img
          alt="darkLogo"
          class="darkLogo"
          src="darkLogo.JPG"
        />
      </a>
    </header>
    <div
      class="cookbook-page-container"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      >
        Welcome to the Cookbook. Here is a list of your potion recipes
      </h1>
      <div
        class="recipes-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
